drivers/include/nrf24l01p.h

```diff
@@ -1,627 +1,1366 @@
+ *          for pipe 1
+ *
+ * @param[in] dev           NRF24L01P device handle
+ * @param[in] payload       Payload
+ * @param[in] payload_width Payload width
+ *
+ * @return                  @see nrf24l01p_set_ack_payload
+ */
+static inline int nrf24l01p_set_ack_payload_p1(nrf24l01p_t *dev,
+                                               const void *payload,
+                                               size_t payload_width)
+{
+    return nrf24l01p_set_ack_payload(dev, payload, payload_width, NRF24L01P_P1);
+}
+
+/**
+ * @brief   Wrapper around @see nrf24l01p_set_ack_payload to write ACK paylaod
+ *          for pipe 2
+ *
+ * @param[in] dev           NRF24L01P device handle
+ * @param[in] payload       Payload
+ * @param[in] payload_width Payload width
+ *
+ * @return                  @see nrf24l01p_set_ack_payload
+ */
+static inline int nrf24l01p_set_ack_payload_p2(nrf24l01p_t *dev,
+                                               const void *payload,
+                                               size_t payload_width)
+{
+    return nrf24l01p_set_ack_payload(dev, payload, payload_width, NRF24L01P_P2);
+}
+
+/**
+ * @brief   Wrapper around @see nrf24l01p_set_ack_payload to write ACK paylaod
+ *          for pipe 3
+ *
+ * @param[in] dev           NRF24L01P device handle
+ * @param[in] payload       Payload
+ * @param[in] payload_width Payload width
+ *
+ * @return                  @see nrf24l01p_set_ack_payload
+ */
+static inline int nrf24l01p_set_ack_payload_p3(nrf24l01p_t *dev,
+                                               const void *payload,
+                                               size_t payload_width)
+{
+    return nrf24l01p_set_ack_payload(dev, payload, payload_width, NRF24L01P_P3);
+}
+
+/**
+ * @brief   Wrapper around @see nrf24l01p_set_ack_payload to write ACK paylaod
+ *          for pipe 4
+ *
+ * @param[in] dev           NRF24L01P device handle
+ * @param[in] payload       Payload
+ * @param[in] payload_width Payload width
+ *
+ * @return                  @see nrf24l01p_set_ack_payload
+ */
+static inline int nrf24l01p_set_ack_payload_p4(nrf24l01p_t *dev,
+                                               const void *payload,
+                                               size_t payload_width)
+{
+    return nrf24l01p_set_ack_payload(dev, payload, payload_width, NRF24L01P_P4);
+}
+
+/**
+ * @brief   Wrapper around @see nrf24l01p_set_ack_payload to write ACK paylaod
+ *          for pipe 5
+ *
+ * @param[in] dev           NRF24L01P device handle
+ * @param[in] payload       Payload
+ * @param[in] payload_width Payload width
+ *
+ * @return                  @see nrf24l01p_set_ack_payload
+ */
+static inline int nrf24l01p_set_ack_payload_p5(nrf24l01p_t *dev,
+                                               const void *payload,
+                                               size_t payload_width)
+{
+    return nrf24l01p_set_ack_payload(dev, payload, payload_width, NRF24L01P_P5);
+}
```

Comment:
Will the user really need those functions? Otherwise it is better to split those up to an internal header. This makes life much easier later, as internal APIs can easily be changed. For user facing APIs we have a deprecation process that can be quite painful during development.

Reply:
The possibility to transmit payload in an ACK is not useful for RIOT. Because you must set the payload before you get the request, because SPI write would be to slow.
(Knowing the answer before you hear the question.)
But since the device supports such ACK payloads, I wanted to implement it.

Or did you mean that I should just drop the `static inline` wrappers?

Reply:
> Or did you mean that I should just drop the `static inline` wrappers?

No, they have no cost and make the code using those functions more compact.

There is value in exposing this feature to the use, so it makes sense to keep it in the user facing API.